src/openrct2/paint/paint.c

```diff
@@ -733,7 +733,35 @@ void paint_generate_structs(rct_drawpixelinfo * dpi)
     }
 }
 
-static paint_struct * paint_arrange_structs_helper(paint_struct * ps_next, uint16 quadrantIndex, uint8 flag)
+static bool compareSpriteBBoxes(uint8 rotation, const paint_struct_bound_box *initialBBox,
```

Comment:
I would probably call this something like isBBoxIntersecting as its not related to any sprite.